Realm/RLMRealmConfiguration.mm

```diff
@@ -275,4 +279,21 @@ - (NSString *)pathOnDisk {
     return @(_config.path.c_str());
 }
 
+- (void)setShouldCompactOnLaunch:(RLMShouldCompactOnLaunchBlock)shouldCompactOnLaunch {
+    _shouldCompactOnLaunch = shouldCompactOnLaunch;
```

Comment:
Should set this after validation so that the object isn't left in an inconsistent state when validation fails.